### PR TITLE
CONSOLE-4182: replace DropdownDeprecated in SearchFilterDropdown

### PR DESCRIPTION
**Pull request by rhamilto**
After

https://github.com/user-attachments/assets/1da27c1d-2f1d-4b92-9743-62f0e16ee1db

### CODE REVIEW

**Comment by rhamilto** [comment]
/hold

Need to fix accessibility issues in integration tests

**Comment by rhamilto** [comment]
/hold cancel

Integration tests fixed.  @TheRealJon, mind re-tagging?

**Comment by rhamilto** [comment]
> Integration tests fixed. @TheRealJon, mind re-tagging?

Er, reviewing.

File: frontend/public/components/search-filter-dropdown.tsx
Comment (logonoff) [inline frontend/public/components/search-filter-dropdown.tsx:91]: you might make these ids unique (e.g., through using [_.uniqueId](https://lodash.com/docs/4.17.15#uniqueId)) as there could be a problem with this aria label being invalid due to multiple `#toggle-id`s present in the dom

Comment (rhamilto) [reply frontend/public/components/search-filter-dropdown.tsx]: Good suggestion, @logonoff.  Updated.  

`toggle-id` was the existing value, but a quick search shows [we're using that same value elsewhere in the codebase](https://github.com/search?q=repo%3Aopenshift%2Fconsole+id%3D%22toggle-id%22&type=code).

**Comment by rhamilto** [comment]
/hold 

for https://github.com/openshift/console/pull/14394

File: frontend/public/components/search-filter-dropdown.tsx
Comment (logonoff) [inline frontend/public/components/search-filter-dropdown.tsx:57]: one more nit: i don't think these react.fragments are necessary anymore since the entire component is wrapped by an `InputGroup`

**Review by logonoff — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
QE Approver
/assign @yapei

Tech debt, so manually assigning labels
/label px-approved
/label docs-approved

**Comment by yanpzhan** [comment]
Tested on cluster launched against the pr, on Home->Search page, checked filter dropdown list, tried filter with name/label, it worked. There is not regression issue.
/label qe-approved

**Pull request merged**
